YurisCodingClub/pro-check
# Added `GitHub Repo` checklist with `localStorage` functionality

## Description



- Added `GitHub.js` component to `src/componts` folder
- Import `GitHub` to `App.js` component
- Create an Array of Object with id, checked, and item properties and write all the tasks given in the image of the issue
- Add `Local Storage` functionality to save the checked/unchecked items
- Used HTML Semantic tags for better accessibility
- Used `dangerouslySetInnerHTML` to render HTML inside of a string property

## Related Issue


#3 

## Acceptance Criteria



- [x] Create a component called GitHub.js and save it in a folder called components in the src directory
- [x]  Heading: GitHub Repositories
- [x] Input fields must be editable
- [x] Input fields are [accessible](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/form)
- [x] No stylings but plain html
- [x] The new component is imported and displayed in the App.js component
- [x]  Only the new component is displayed
- [x]  Save data in localStorage


## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
| ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before


![image](https://user-images.githubusercontent.com/47440165/200181784-f2467536-3658-4f01-86c5-ccfc0fa943e3.png)



### After


![image](https://user-images.githubusercontent.com/47440165/200181842-3e0935d7-9c24-40f5-a7a3-6b39057bdd51.png)


## Testing Steps / QA Criteria

- To check the code visit `src/components/GitHub.js`
- Clear your `Local Storage`
- After clearing `Local Storage` , visit the preview link to test live.
